cli/src/lib/pwa/compileServiceWorker.js

```diff
@@ -0,0 +1,96 @@
+const path = require('path')
+const { reporter } = require('@dhis2/cli-helpers-engine')
+const webpack = require('webpack')
+const getPWAEnvVars = require('./getPWAEnvVars')
+
+/**
+ * Uses webpack to bundle a service worker. If used in development mode,
+ * compiles in development mode and outputs the service worker to the `public`
+ * dir for use with a dev server. In production mode, compiles a minified
+ * service worker and outputs it into the apps `build` dir.
+ *
+ * Currently used only for 'dev' SWs, since CRA handles production bundling.
+ * TODO: Use this for production bundling as well, which gives greater control
+ * over 'injectManifest' configuration (CRA omits files > 2MB) and bundling
```

Comment:
Yeah, maybe! Once the file is precached, it should be pretty performant to load them from the cache, right? So the costly download would happen only happen on first load.

I first noticed this when I built the dashboard app using the PWA branches, where one of the chunks was omitted from the precache manifest because it was > 2MB